module-1/lab-errhand_listcomp/your-code/main_miriam.py

```diff
@@ -7,44 +7,53 @@
 print(my_listComprehension)
```

Comment:
Question 7: you don't need to make a join to solved the problem. Remember to use always the print statement whether it is a Jupyter notebook or a py script. The example shown in class: 
```Python
vowels = ['a','e','i','o','u',' ']
non_vowels = [letter for letter in teststring if letter.lower() not in vowels]
print(non_vowels)
```
Question 8: You don't need to use join and to list all the capital letters. There are 2 useful functions for these cases, isupper(), islower(). Another way: 
```Python
results = [letter for letter in teststring if letter.isupper()]
print(capital_letters)
print(results)
```
Question 9: When you need to look up for list of many items don't write them all in a string. This can cause unexpected operation issues and it makes your code less readable for you and others. An example: 
```Python
teststring = 'The quick brown fox jumped over the lazy dog'
vowels = ['a','e','i','o','u',' ']
consonants = [letter for letter in teststring if letter.lower() not in vowels]
print(consonants)
```
Question 10: Remember to use always relative paths instead of absolute paths. The example shown in class:
```Python
import os
from os import walk
mypath = '../madrid-oct-2018'
files = [folder for folder in next(os.walk(mypath))[1]]
print(files)
``` 
Question 11: List comprehensions are there for make our live easier. Keep code as simple as possible and, preferably, in a single readable line. A few examples: 
```Python
#One way
random_lists = [random.sample(range(101),10) for _ in range (4)]
print(random_lists)
#Other way (AlbertoG's)
res = [[random.randint(1,101) for _ in range(10)] for _ in range(4)]
print(res)
#Another way (Ivan's)
random_lists=[random.sample(range(1, 100), 10) for f in range(4)]
print(random_lists)
```
Question 12: You can also use flatten() function from numpy library as follows: 
```Python
#AlbertoG's approach 
import numpy as np
flatten_list = np.array(list_of_lists).flatten()
```
 Question 15: What's this zero_excepcion() for? You didn't define a function, did you?